lib/index.js

```diff
@@ -1,64 +1,39 @@
 'use strict';
 var App = require('./app');
 var _ = require('lodash');
-
-var S3rver = function (options) {
-  this.options = {
-    port: 4578,
-    hostname: 'localhost',
-    silent: false,
-    indexDocument: '',
-    errorDocument: '',
-    fs: require('fs-extra')
-  };
-
-  if (options) {
-    this.options = _.merge(this.options, options);
+var https = require('https');
+
+var defaultOptions = {
+  port: 4578,
+  hostname: 'localhost',
+  directory: '.',
+  silent: false,
+  indexDocument: '',
+  errorDocument: '',
+  fs: require('fs-extra')
+};
+
+function S3rver(options) {
+  this.options = _.merge({}, defaultOptions, options);
+  this.app = new App(this.options);
+}
+
+for (var option in defaultOptions) {
```

Comment:
I do not like this for loop for dynamically adding methods.

Reply:
I actually don't really see the point in having all those setters in the first place. I guess I just felt better by sticking them all in a loop so they weren't wasting so much space. I'll leave them alone when I fix this up.

Reply:
Actually we can drop those since they can be set as the options of the constructor anyway... 😅